src/validator.cpp

```diff
@@ -351,21 +351,35 @@ void Validator::validateModel(const ModelPtr &model)
                         addIssue(issue);
                         foundImportIssue = true;
                     }
-                    // Check for a xlink:href.
-                    // TODO: check this id against the XLink spec (see CellML Spec 5.1.1).
+                    // Check for a xlink:href and its format.
                     if (importSource.empty()) {
                         IssuePtr issue = Issue::create();
                         issue->setDescription("Import of units '" + unitsName + "' does not have a valid locator xlink:href attribute.");
                         issue->setImportSource(units->importSource());
                         issue->setReferenceRule(Issue::ReferenceRule::IMPORT_HREF);
                         addIssue(issue);
                         foundImportIssue = true;
+                    } else {
+                        xmlURIPtr uri = xmlParseURI(importSource.c_str());
+                        if (uri == nullptr) {
+                            IssuePtr issue = Issue::create();
+                            issue->setDescription("Import of units '" + unitsName + "' has an invalid URI in the href attribute.");
```

Comment:
I would remove "in the href attribute". `href` is indeed an XLink attribute, which a "normal" libCellML user shouldn't have to know about.

Reply:
This means that we might have/want to rename `Issue::ReferenceRule::IMPORT_HREF`?

Reply:
The pattern established in PR #604 is item_attribute so I don't understand what's wrong with `IMPORT_HREF`?

Reply:
And the error follows that in the existing error above which uses `xlink:href`?

Reply:
Just the fact that it contains `HREF`, but I understand that it's an enum, hence I was merely asking a question.

Reply:
Otherwise, the error message should read "xlink:href" and not "href", i.e. as in line 357.

Reply:
Also, it would be nice to be consistent in our error messages. Line 357 reads "...does not have a valid..." while here we have "... has an invalid..."

Reply:
There's a difference between the presence of something invalid (the second error) and the absence of anything, valid or not (the first error).  We don't want to be consistent because they're not talking about the same thing.

Reply:
"xlink:" added in 9adc062e.